src/lib/tvgSceneImpl.h

```diff
@@ -44,12 +44,14 @@ struct Scene::Impl
         return true;
     }
 
-    bool update(RenderMethod &renderer, const RenderTransform* transform, RenderUpdateFlag flag)
+    void* update(RenderMethod &renderer, const RenderTransform* transform, vector<Composite>& compList, RenderUpdateFlag flag)
     {
+        void *edata = nullptr;
         for(auto paint: paints) {
-            if (!paint->pImpl->update(renderer, transform, static_cast<uint32_t>(flag))) return false;
+            edata = paint->pImpl->update(renderer, transform, compList, static_cast<uint32_t>(flag));
```

Comment:
Bad.... It looks the scene must return the list of children engine data...
You can handle it like this:

bool update(RenderMethod &renderer, const RenderTransform* transform, vector<Composite>& compList, vector<void*>& edataList, RenderUpdateFlag flag):

Otherwise, please add FIXME: for handle this later.

Reply:
I changed update() return type void* to bool.
Instead, an argument that receives edata was added.